uarray/tests/test_uarray.py

```diff
@@ -255,3 +253,44 @@ def __eq__(self, other):
             nullary_mm()
 
     assert e.value is foo
+
+
+def test_getset_state(cleanup_backends):
+    ua.set_global_backend(Backend())
+    ua.register_backend(Backend())
+
+    with ua.set_backend(Backend()), ua.skip_backend(Backend()):
+        state = ua.get_state()
+
+    pstate = state._pickle()
+
+    assert pstate != ua.get_state()._pickle()
+
+    ua.set_state(state)
```

Comment:
IMO this should be a context manager so that the local context is preserved for callers. Otherwise this could be pretty dangerous. e.g.
```py
def evil_function():
    ua.set_state(foo)
    # Call some multi methods

# Somewhere else
with ua.set_backend(bar):
    evil_function()
    # We have no idea what state we're in
    # ua.set_backend will probably fail in __exit__
```